### PR TITLE
Clean up the top-level organization in the docs

### PR DESCRIPTION
**Pull request by asmeurer**
- Use better summaries of the top level sections on the index page based on
  Diataxis.

- Move the contributing guides into their own section. 

- Potentially "getting started" can also be its own section, although right now it's just the
  installation instructions.

- Made tutorials a section which can hold many tutorials. The current
  introductory tutorial is just one such tutorial. We may also want to split
  it into many tutorials.

- Renamed "explanation" into "explanations" (but I did not change the
  directory name).

- Redirects were added for all moved pages.

- Fix some existing redirects. The redirect paths are supposed to be relative
  to the redirected page. The reredirects extension does not test if a
  redirect is valid.

I haven't looked at the organization of the API sections yet. I will probably do that in a separate PR since it can be done independently.

#### Release Notes

- other
  - Clean up the top-level organization in the documentation.

### CODE REVIEW

**Comment by bertiewooster** [comment]
Seems logical. Two issues:

- Tutorials is missing from left navbar.

missing-tutorials-in-left-navbar

- The Tutorials link in the body of the home page leads to Matplotlib's tutorials page https://matplotlib.org/stable/tutorials/index.html#tutorials. So can't review Tutorials page yet. (I believe something like this happened before; how is SymPy even linked to Matplotlib in terms of documentation?)

**Comment by asmeurer** [comment]
It's linking to matplotlib because we have that set up as an intersphinx mapping. Clearly we need to make that a little more restrictive. The page didn't show up because I forgot to `git add` it. Should be fixed now.

**Comment by moorepants** [comment]
How do you view the circleci preview build? I don't see a URL anywhere obvious.

**Comment by moorepants** [comment]
I see this:
image

Then I click "Details", but I don't see where I would find a URL to click to view the docs.

**Comment by moorepants** [comment]
Duh... it says "click here to see a preview..."

**Comment by bertiewooster** [comment]
Confirmed that fixed the two issues I raised. New tutorials top-level page, currently listing just the introductory tutorial, seems fine.

**Comment by asmeurer** [comment]
Circle itself is mostly useful if you want to see the build log, but if you do end up there, the HTML pages are in the "artifacts" section.

**Comment by oscarbenjamin** [comment]
This looks good to me

**Comment by asmeurer** [comment]
Any thoughts on the installation instructions? Should they stay where they are now (in the how-to guides) or moved somewhere else?

**Comment by oscarbenjamin** [comment]
The 3.9 doctests failed again. We need to figure out the bug in `sympy.combinatorics.perm_groups.PermutationGroup.schreier_sims_random`

**Comment by asmeurer** [comment]
I was also going to look into using cards from https://sphinx-design.readthedocs.io/en/furo-theme/cards.html to organize the categories on the main page, similar to the [pandas documentation](https://pandas.pydata.org/docs/). Do people think that would be a good idea?

**Comment by oscarbenjamin** [comment]
Installation should probably be the first thing at top level before tutorials.

**Comment by asmeurer** [comment]
Right now we have "getting started" but it's just a simple page that then links to the installation. Do you see any reason not to just merge this into one page? Is there anything else we would ever want to add to that page?

**Comment by oscarbenjamin** [comment]
I'm not sure but I think "installation" (at least basic instructions) should be literally the first section of the docs.

**Comment by asmeurer** [comment]
I moved "Installation" to the top-level, and moved everything into cards. I'm still playing around with the [cards extension](https://sphinx-design.readthedocs.io/en/furo-theme/cards.html) so any suggestions for improvements are welcome.

File: doc/src/contributing/index.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.. _contributing:
+
+==============
+ Contributing
+==============
+
+The contributing guide goes over the details necessary to contribute to SymPy.
+See also the full `Development Workflow
```
Comment (bertiewooster) [inline doc/src/contributing/index.rst:8]:
```suggestion
Also refer to the full `Development Workflow
```

to be more inclusive, for example of people using screen readers.

**Comment by bertiewooster** [comment]
Visually, I like the layout--it makes clear that Installation and Contributing are in a different bin than the four Diataxis sections (Tutorials, How-to Guides, Explanations, and API Reference).

There is an [accessibility issue with the cards, per Lighthouse](https://googlechrome.github.io/lighthouse/viewer/?psiurl=https%3A%2F%2Foutput.circle-artifacts.com%2Foutput%2Fjob%2F2fcb6c2d-6976-42b1-b915-7e0aad93ba42%2Fartifacts%2F0%2Fdoc%2F_build%2Fhtml%2Findex.html&strategy=mobile&category=performance&category=accessibility&category=best-practices&category=seo&category=pwa&utm_source=lh-chrome-ext):

```
Links do not have a discernible name
Link text (and alternate text for images, when used as links) that is discernible, unique, and focusable improves the navigation experience for screen reader users.
```

**Comment by asmeurer** [comment]
I'm not sure how to fix that. Seems like it would be an upstream issue with the cards extension. Maybe we can ask them about it https://github.com/executablebooks/sphinx-design/issues

**Comment by bertiewooster** [comment]
I submitted an issue https://github.com/executablebooks/sphinx-design/issues/78.

**Comment by oscarbenjamin** [comment]
This looks a lot better. Leaving aside the organisation of the API reference for now some other issues I see (that could be fixed in another  PR):

The assumptions "guide" probably belongs in explanation rather than guides.

Maybe deprecations should be moved from explanation to API reference?

The tutorials page only has one item. Are we actually going to add more tutorials there or should we just consider each of the subsections of the tutorial to be a separate tutorial and list them all directly?

The section "SymPy Special Topics" under explanation is a bit weird. Maybe move the classification section out to be its own explanation page. The "finite difference" page doesn't seem like it belongs anywhere but is probably more suited to guides than explanation.

**Comment by oscarbenjamin** [comment]
This has merge conflicts now

**Comment by oscarbenjamin** [comment]
> Any additional comments on this?

Not from me. I think it looks good. My comments above were really suggestions for future PRs. I thought that this was good to merge before.

**Comment by bertiewooster** [comment]
My only concern is the accessibility issue with the cards on the homepage: because they lack a discernible name, visitors using screen readers will have their access to the entire site severely impeded. Unfortunately, the [Executable Books project has not addressed this accessibility issue](https://github.com/executablebooks/sphinx-design/issues/78). Because the cards are a severe impedance for visitors using screen readers, and just a nice visual flourish for sighted visitors, I strongly believe we should remove the cards construct until discernible names can be attached to those links.

**Comment by asmeurer** [comment]
I moved the cards commit to https://github.com/sympy/sympy/pull/23746. This should now be mergeable as just the reorganization.

**Comment by oscarbenjamin** [comment]
There was a problem with CI and I guess something else got merged in the mean time because this has conflicts now.

**Comment by asmeurer** [comment]
Ugh. Let's try again. The annoying thing is that git doesn't actually give merge conflicts with it does the merge. I think GitHub must be using an older version of git that doesn't have as smart of merge conflict resolution.

**Comment by oscarbenjamin** [comment]
I think this went in after the 1.11 branch was created so it would need to be backported to 1.11.

**Pull request merged**